### ISSUE TITLE
Can I keep only lip sync? How can I cancel head movements and blinks?

### ISSUE THREAD

**Issue by BInyLU** [deepbrainai-research/float#12]
Can I keep only lip sync? How can I cancel head movements and blinks?

**Comment by TaekyungKi** [comment]
Hi @BInyLU,
The answer is no. With this repo, you can only generate a whole head motion (including lips and head movement).
FLOAT framework actually is capable of  separately generating them using 3DMM parameters as reported in the paper.
But, this repo only supports the whole head motion generation.
Sorry for your inconvenience.

**Comment by TaekyungKi** [comment]
Hi @jupinter,
Yes, exactly. I recommend to use a linear layer (dim 6 -> 6) followed by the layer norm.

**Comment by jupinter** [comment]
Thanks for your reply. 
I would like to confirm some more details. 
Do I just need to concatenate the 3DMM pose parameters to the back of the sample? Should each frame now have 512 dimensions?

`
if a_cfg_scale is None: a_cfg_scale = self.opt.a_cfg_scale
if r_cfg_scale is None: r_cfg_scale = self.opt.r_cfg_scale
if e_cfg_scale is None: e_cfg_scale = self.opt.e_cfg_scale
sample = self.sample(data, a_cfg_scale = a_cfg_scale, r_cfg_scale = r_cfg_scale, e_cfg_scale = e_cfg_scale, emo = emo, nfe = nfe, seed = seed)
`
Looking forward to your response. Thank you very much!

**Comment by TaekyungKi** [comment]
Hi @jupinter,
Yes, you just need to concatenate the 6-dim poses along with the other driving signals (so the dimension of the driving conditions will increase), and then you project them into the hidden dimension in each FMT block.
One thing I'd like to remind you is that this repository only supports the audio-driven motion generation.
So if you want to inject the head pose, basically you have to train FMT from scratch (or apply some training-free injection).
In my case, the pose cfg scale, `p_cfg_scale` 1 is enough for controlling the pose.

**Comment by jupinter** [comment]
Hi , @TaekyungKi 

Retraining the FM model is indeed a significant amount of work for me, but I'm wondering if there's a simpler way. I would like to keep the following lines unchanged:

```python
c = torch.cat([wr, wa, we], dim=-1)
c = self.c_embedder(c)
```

The output dimension of `self.c_embedder` is 1024. So I plan to add `self.rt_embedder` with input and output dimensions of 6 and 1024, respectively. Then, I will concatenate the two outputs and only train the parameters of `self.rt_embedder`, keeping the other parameters unchanged. Do you think this is feasible?

Best regards!
 Jupinter

**Comment by jupinter** [comment]
@TaekyungKi  Thank you very much for you reply!
